cmd/erasure-server-pool.go

```diff
@@ -1980,6 +1980,24 @@ func (z *erasureServerPools) DeleteBucket(ctx context.Context, bucket string, op
 		defer lk.Unlock(lkctx)
 	}
 
+	if !opts.Force {
+		results := make(chan itemOrErr[ObjectInfo])
+		go z.Walk(ctx, bucket, "", results, WalkOptions{Limit: 1})
```

Comment:
```suggestion
		go z.Walk(ctx, bucket, "", results, WalkOptions{Limit: 1})
```

pass a context which can canceled upon return, don't wait on client context to cancel this goroutine.